src/anyp/Uri.h

```diff
@@ -179,7 +179,13 @@ operator <<(std::ostream &os, const AnyP::Uri &url)
 class HttpRequest;
 
 void urlInitialize(void);
-char *urlCanonicalClean(const HttpRequest *);
+/// \returns a pointer to a local static buffer containing request URI
+/// that honors strip_query_terms and %-encodes unsafe URI characters
+char *urlCanonicalClean(const HttpRequest &);
```

Comment:
Since you are touching most if not all the callers of this old function anyway please take the opportunity to just remove it in favour of your improved *WithoutRequest() function. The less dependencies on HttpRequest symbols we have from this Uri.cc code the better off Squid is.